src/core/cardano/walletConnect/identityWalletConnect.ts

```diff
@@ -0,0 +1,112 @@
+import { Buffer } from "buffer";
+import {
+  Paginate,
+  Cip30DataSignature,
+  IWalletInfo,
+} from "@fabianbormann/cardano-peer-connect/dist/src/types";
+import { CardanoPeerConnect } from "@fabianbormann/cardano-peer-connect";
+import { AriesAgent } from "../../agent/agent";
+import { IdentifierType } from "../../agent/services/identifierService.types";
+
+class IdentityWalletConnect extends CardanoPeerConnect {
+  getIdentifierId: () => Promise<string>;
+  signDataWithIdentifier: (
+    identifierId: string,
+    payload: string
+  ) => Promise<string>;
+  generateOobi: (identifierId: string) => Promise<string>;
+
+  constructor(
+    walletInfo: IWalletInfo,
+    seed: string | null,
+    announce: string[],
+    discoverySeed?: string | null
+  ) {
+    super(walletInfo, {
+      seed: seed,
+      announce: announce,
+      discoverySeed: discoverySeed,
+      logLevel: "info",
+    });
+
+    this.getIdentifierId = async (): Promise<string> => {
```

Comment:
Do these 3 functions need to be defined in the constructor? Or is this a restriction of how peer connect works? Seems weird.

Reply:
When you start the [peerConnect function to enable to experimental api you need to send the functions as objects](https://github.com/fabianbormann/cip-0045-demo-implementation/blob/7e7a24018f64d8b55b8a567ba51ffa5e17efe442/demo-wallet-app/src/DemoWalletConnect.tsx#L68). This is the most elegant way that I found to do this. Is there is any other way let me know :)

Reply:
Interesting, but does I thought if `getIdentifierId` was a normal method in a class, that you can still access it from the instance of the class with `this.identityWalletConnect.getIdentifierId`, no?

Reply:
OK, this is OK then, no worries!